libraries/AP_TemperatureSensor/AP_TemperatureSensor_MCP9600.cpp

```diff
@@ -129,13 +126,18 @@ bool AP_TemperatureSensor_MCP9600::set_config(const ThermocoupleType thermoType,
 
 bool AP_TemperatureSensor_MCP9600::read_temperature(float &temperature)
```

Comment:
Could I suggest:
```
bool AP_TemperatureSensor_MCP9600::read_temperature(float &temperature)
{
    // confirm new temperature is available and then read it

    // First, read STATUS register:
    uint8_t data[2];
    if (!_dev->read_registers(MCP9600_CMD_STATUS, data, 1)) {
        return false;
    }
    // check UPDATE bit for new temperature data ready:
    if ((data[0] & MCP9600_CMD_STATUS_UPDATE_READY) == 0) {
        return false;
    }
    // clear update bit:
    if (!_dev->write_register(0x04, data[0] & ~MCP9600_CMD_STATUS_UPDATE_READY)) {
        return false;
    }
    // read temperature:
    if (!_dev->read_registers(MCP9600_CMD_HOT_JUNCT_TEMP, data, 2)) {
        return false;
    }

    // scale temperature:
    temperature = int16_t(UINT16_VALUE(data[0], data[1])) * AP_TemperatureSensor_MCP9600_SCALE_FACTOR;

    return true;
}
```

Reply:
yeah.. I started doing it that way.. but it seemed excessive. No issue changing it to that in the name of readability

Reply:
I've applied this suggestion